src/Streamly/System/Process.hs

```diff
@@ -172,56 +169,88 @@ wait procHandle = liftIO $ do
         ExitFailure exitCodeInt -> throwM $ ProcessFailure exitCodeInt
 
--- |
--- Creates a process using the path to executable and arguments, then
--- builds three pipes, one whose read end is made the process's standard
--- input, and another whose write end is made the process's standard
--- output, and the final one's write end is made the process's standard error.
--- The function returns the handle to write end to the process's input, handle
--- to read end to process's output, handle to read end to process's standard
--- error and process handle of the process
---
-openProcErr ::
-    FilePath                                -- ^ Path to Executable
-    -> [String]                             -- ^ Arguments
+-- Set everything explicitly so that we are immune to changes upstream.
+procAttrs :: FilePath -> [String] -> CreateProcess
+procAttrs path args = CreateProcess
+    { cmdspec = RawCommand path args
+    , cwd = Nothing -- inherit
+    , env = Nothing -- inherit
+
+    -- File descriptors
+    , std_in = CreatePipe
+    , std_out = CreatePipe
+    , std_err = Inherit
+    , close_fds = True  -- False?
+
+    -- Session/group/setuid/setgid
+    , create_group = False
+    , child_user = Nothing  -- Posix only
+    , child_group = Nothing  -- Posix only
+
+    -- Signals (Posix only) behavior
+    -- Reset SIGINT (Ctrl-C) and SIGQUIT (Ctrl-\) to default handlers.
+    , delegate_ctlc = False
+
+    -- Terminal behavior
+    , new_session = False  -- Posix only
+    , detach_console = False -- Windows only
+    , create_new_console = False -- Windows Only
+
+    -- Added by commit 6b8ffe2ec3d115df9ccc047599545ca55c393005
+    , use_process_jobs = True -- Windows only
+    }
+
+pipeStdErr :: CreateProcess -> CreateProcess
+pipeStdErr cfg = cfg { std_err = CreatePipe }
+
+-- | Creates a system process from an executable path and arguments. For the
+-- default attributes used to create the process see 'procAttrs'.
+--
+createProc' ::
+       (CreateProcess -> CreateProcess) -- ^ Process attribute modifier
+    -> FilePath                         -- ^ Executable path
+    -> [String]                         -- ^ Arguments
     -> IO (Handle, Handle, Handle, ProcessHandle)
     -- ^ (Input Handle, Output Handle, Error Handle, Process Handle)
-openProcErr path args = do
-    let spec = proc path args
-        spec1 =
-            spec
-                { std_in = CreatePipe
-                , std_out = CreatePipe
-                , std_err = CreatePipe
-                , close_fds = True
-                , use_process_jobs = True
-                }
-
+createProc' modCfg path args = do
     (Just stdinH, Just stdoutH, Just stderrH, procH)
-        <- createProcess spec1
+        <- createProcess $ modCfg $ procAttrs path args
     return (stdinH, stdoutH, stderrH, procH)
 
-{-# INLINE processChunks #-}
-processChunks ::
+{-# INLINE runStdin #-}
+runStdin :: (MonadIO m, IsStream t) => t m (Array Word8) -> Handle -> t m a
```

Comment:
How did you come up with this signature?
This is not specific to stdin, this is very general. It puts the contents of a stream into a handle. You should probably rename it.
`putChunksToHandle`

Reply:
It was a local function earlier now factored out because its used at two places. Just added the signature because it is top level now. 

It returns a nil stream so can match any output stream type `t m a`.

I think its clearer to keep it named as runStdin, because it makes it clear how we are handling stdin. Also, its consistent with runStdout and runStderr.

Reply:
You should also make something general with `after`, to replace `stdErr` and `stdOut` I guess.

Reply:
Or you can just ignore it. Look good enough.